vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -4416,9 +4417,9 @@ public void scrollToStart() {
      * Scrolls to the last data row of the grid.
      */
     public void scrollToEnd() {
-        getUI().ifPresent(
-                ui -> ui.beforeClientResponse(this, ctx -> getElement()
-                        .executeJs("this.scrollToIndex(this._effectiveSize)")));
+        getUI().orElseGet(UI::getCurrent).beforeClientResponse(this,
```

Comment:
How about using `getElement().getNode().runWhenAttached` to run the code when the component is actually added to a UI? There's a few other places in the component that use this approach.